netket/jax/utils.py

```diff
@@ -304,7 +305,10 @@ def PRNGKey(
     The same seed will be distributed to all processes.
     """
     if seed is None:
-        key = jax.random.PRNGKey(random_seed())
+        seed = random_seed()
+        if config.netket_experimental_pjit and jax.process_count() > 1:
+            seed = jax.experimental.multihost_utils.broadcast_one_to_all(seed)
```

Comment:
shouldn't this go below, at the same level of the MPI call?
otherwise one might end up with different seeds on different nodes?

Reply:
The only think I wanted to make sure is that all the processes have the same seed if we use the default one which comes from numba. It's not strictly necessary, but this way we have the same random state on all of them, so it's easier to reproduce results
In any other case we require the user to run the same program on every process, and give us the same seed/inital key.
Jax then knows how many devices and processes there are and generates a different random number depending on the process and device, just like it would be if it was a single one, given the same key.

Also note that we will [not be able to distribute keys in the future anyway](https://github.com/google/jax/pull/16511#issuecomment-1603923832).

Reply:
food point. thanks Clemens.